src/legacy/core_plugins/kibana/public/discover/kibana_services.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import 'ui/collapsible_sidebar';
+import 'ui/directives/listen';
+import 'ui/fixed_scroll';
+import 'ui/directives/css_truncate';
+
+import { npStart } from 'ui/new_platform';
+import chromeLegacy from 'ui/chrome'; // just used in embeddables
+import angular from 'angular'; // just used in embeddables and discover controller
+import uiRoutes from 'ui/routes';
+// @ts-ignore
+import { uiModules } from 'ui/modules';
+import { SearchSource } from 'ui/courier';
+// @ts-ignore
+import { StateProvider } from 'ui/state_management/state';
+// @ts-ignore
+import { SavedObjectProvider } from 'ui/saved_objects/saved_object';
+import { SavedObjectRegistryProvider } from 'ui/saved_objects/saved_object_registry';
+import { FilterBarQueryFilterProvider } from 'ui/filter_manager/query_filter';
+import { timefilter } from 'ui/timefilter';
+import { ShareContextMenuExtensionsRegistryProvider } from 'ui/share';
+// @ts-ignore
+import { IndexPattern, IndexPatterns } from 'ui/index_patterns';
+import { wrapInI18nContext } from 'ui/i18n';
+// @ts-ignore
+import { docTitle } from 'ui/doc_title';
+// @ts-ignore
+import * as docViewsRegistry from 'ui/registry/doc_views';
+
+const services = {
+  // new plattform
+  capabilities: npStart.core.application.capabilities,
+  chrome: npStart.core.chrome,
+  docLinks: npStart.core.docLinks,
+  eui_utils: npStart.plugins.eui_utils,
+  inspector: npStart.plugins.inspector,
+  metadata: npStart.core.injectedMetadata.getLegacyMetadata(),
+  toastNotifications: npStart.core.notifications.toasts,
+  uiSettings: npStart.core.uiSettings,
+  // legacy
+  chromeLegacy,
+  docTitle,
+  docViewsRegistry,
+  FilterBarQueryFilterProvider,
+  SavedObjectRegistryProvider,
+  SavedObjectProvider,
+  SearchSource,
+  ShareContextMenuExtensionsRegistryProvider,
+  StateProvider,
+  timefilter,
+  uiModules,
+  uiRoutes,
+  wrapInI18nContext,
```

Comment:
with separating services + static dependencies,  it's much better now, do you see some further improvements dear @flash1293

Reply:
Looks pretty good to me 👍